yjit/src/codegen.rs

```diff
@@ -547,7 +546,7 @@ pub fn gen_counted_exit(side_exit: CodePtr, ocb: &mut OutlinedCb, counter: Optio
 
     // Jump to the existing side exit
     asm.jmp(Target::CodePtr(side_exit));
-    asm.compile(ocb, None);
+    let _ = asm.compile(ocb, None); // Caller checks cb.has_dropped_bytes()
```

Comment:
I'm actually not sure if this is safe. For example, `jit_ensure_block_entry_exit` -> `get_side_exit` -> `gen_counted_exit` doesn't seem to check `has_dropped_bytes()`.

Can't we just propagate `Option<CodePtr>` to the callers (i.e. this function should also return `Option<CodePtr>`)? It seems like the point of this PR.

Also, I thought you were going to let `asm.compile()` return `CodePtr` as well. If it returns `Option<(CodePtr, Vec<u32>)>`, you would map it to `.0` of it, always respecting the `Option` state instead of discarding it with `let _ =`. Using `ocb.get_write_ptr()` seems prone to bugs compared to always using an `Option`al return value.

Reply:
Sorry, I definitely cut some corners to keep the diff small. The cases where we ignore the error are probably fine because they're caught by checks at the end of `gen_single_block()`, but propagating the error is definitely better.

I'll go work on a v2.

Reply:
I appreciate you trying to minimize the diff. That makes reviews easier. I'd be comfortable approving the current diff and waiting for another PR on top of it too. I just wasn't sure if you want to fix it since having the `// Caller checks cb.has_dropped_bytes()` comment (instead of a TODO comment) seemed to me like you're saying "this is good". 

If the number of lines wouldn't change that much, it seems fine to do that in this PR too.